langkit/templates/c_api/header_c.mako

```diff
@@ -183,6 +183,34 @@ ${iterator_types.incomplete_decl(T.entity.iterator)}
     % endif
 % endfor
 
+/*
+ * Types for event handler
+ */
+
+${c_doc('langkit.event_handler_type')}
+typedef void *${event_handler_type};
+
+${c_doc('langkit.event_handler_unit_requested_type')}
+typedef void (*${event_handler_unit_requested_type})(
+   void *data,
+   ${analysis_context_type} context,
+   ${text_type} *name,
```

Comment:
Unit names are filenames, and we represent them with bytes in the Ada and C APIs, so this should be `const char *` (like in `get_analysis_unit_from_file`)

Alternatively, IIRC we do create analysis units even when the file is not found, so we could pass directly that new analysis unit (which gives access to the filename) instead of just its file name. What do you think?

Reply:
> Alternatively, IIRC we do create analysis units even when the file is not found, so we could pass directly that new analysis unit (which gives access to the filename) instead of just its file name. What do you think?

The "unit requested" signal is supposed to happen *before* the unit is actually parsed, so I don't think it would make sense passing a unit here.

Reply:
Actually this isn't necessarily a file name here: Since for the moment the langkit user has the job of adding the callbacks, and will decide what to pass here, he could decide to pass the abstract language level name (`Ada.Directories`) or the filename (`a-direct.ads`).

The langkit test uses the first way, and libadalang the second, and depending on the language, I think both could make sense.

> Alternatively, IIRC we do create analysis units even when the file is not found, so we could pass directly that new analysis unit (which gives access to the filename) instead of just its file name. What do you think?

We would lose the ability to pass the abstract name, but we could pass both, and enforce that the `name` parameter must be the abstract name.

But to be honest, while I'm not sure this interface is perfect yet, I need to do a bit of thinking on the LAL side, and maybe a bit of cleaning up in env hooks & all, to see how this all could be simplified, and I'd rather revisit this after if that's OK with you.

Reply:
I see, thanks for the rationale. Could you then add a warning in the doc of this callback? (very briefly describing the probable evolution would also be nice) In any case, enhancing the current doc to give semantics (or the lack of semantics) for this `name` argument looks necessary.

Reply:
The problem with this is that this doc is going to be propagated in the generated bindings, where the semantic is fixed. The most important use case being Libadalang, where it is indeed a file name at the moment.

Reply:
Ah, that’s annoying… we need to fix this eventually: can you leave a TODO?